src/realsense2-all.cmake

```diff
@@ -0,0 +1,118 @@
+                        set_property( GLOBAL PROPERTY _${target_name}_static_bundle_${dependency} ON )
+                        _recursively_collect_dependencies( ${dependency} )
+                    else()
+                        #message( "   x ${dependency} already done" )
+                    endif()
+                else()
+                    #message( "   x ${dependency} not a target" )
+                endif()
+            endforeach()
+        endif()
+        set( static_libs ${static_libs} PARENT_SCOPE )
+    endfunction()
+
+    list( APPEND static_libs ${target_name} )
+    _recursively_collect_dependencies( ${target_name} )
+    list( REMOVE_DUPLICATES static_libs )
+    #message( "---> Dependencies for ${bundle_name}: ${static_libs}" )
+
+    if( CMAKE_CXX_COMPILER_ID MATCHES "^(Clang|GNU)$" )
+
+        set( bundle_path
+                "${CMAKE_ARCHIVE_OUTPUT_DIRECTORY}/${CMAKE_STATIC_LIBRARY_PREFIX}${bundle_name}${CMAKE_STATIC_LIBRARY_SUFFIX}" )
+
+        file( WRITE ${CMAKE_BINARY_DIR}/${bundle_name}.ar.in
+            "CREATE ${bundle_path}\n" )
+        
+        foreach( tgt IN LISTS static_libs )
+            file( APPEND ${CMAKE_BINARY_DIR}/${bundle_name}.ar.in
+                "ADDLIB $<TARGET_FILE:${tgt}>\n" )
+        endforeach()
+    
+        file( APPEND ${CMAKE_BINARY_DIR}/${bundle_name}.ar.in "SAVE\n" )
+        file( APPEND ${CMAKE_BINARY_DIR}/${bundle_name}.ar.in "END\n" )
+
+        file( GENERATE
+            OUTPUT ${CMAKE_BINARY_DIR}/${bundle_name}.ar
+            INPUT ${CMAKE_BINARY_DIR}/${bundle_name}.ar.in )
+
+        set( ar_tool ${CMAKE_AR} )
+        if( CMAKE_INTERPROCEDURAL_OPTIMIZATION )
+            set( ar_tool ${CMAKE_CXX_COMPILER_AR} )
+        endif()
+
+        add_custom_command(
+            COMMAND ${ar_tool} -M < ${CMAKE_BINARY_DIR}/${bundle_name}.ar
+            DEPENDS ${static_libs}
+            OUTPUT ${bundle_path}
+            COMMENT "Bundling ${bundle_path} (with ${static_libs})"
+            VERBATIM )
+
+    elseif( MSVC )
+
+        set( bundle_path  # NOTE: generator expression output in the COMMENT below only works with CMake 3.26
+                "${CMAKE_ARCHIVE_OUTPUT_DIRECTORY}/$<CONFIG>/${CMAKE_STATIC_LIBRARY_PREFIX}${bundle_name}${CMAKE_STATIC_LIBRARY_SUFFIX}" )
+
+        foreach( tgt IN LISTS static_libs )
+            list( APPEND static_libs_full_names $<TARGET_FILE:${tgt}> )
+        endforeach()
+
+        # /IGNORE:4006 - "symbol already defined in <object>; second definition ignored"
+        #       happens because rsutils includes easylogging++, but so does realsense2
+        add_custom_command(
+            COMMAND ${CMAKE_AR} /NOLOGO /IGNORE:4006 /OUT:${bundle_path} ${static_libs_full_names}
+            DEPENDS ${static_libs}
+            OUTPUT ${bundle_path}
+            COMMENT "Bundling ${bundle_path} (with ${static_libs})"
+            VERBATIM )
+
+    else()
+        message( FATAL_ERROR "Unknown bundle scenario!" )
```

Comment:
Are we sure we only support clang/GNU/MSVC?
I see user use Ninja and other compilers as well.
Maybe we should drop the build of realsense-all on that case and not fail the cmake?

Reply:
I'll try to make it so it'll just give a warning and not add realsense2-all.
I can also change it to `if( WIN32 )` ... `else()` ... `endif()`, if you think that's better?

Reply:
WIN32 is better IMO

Reply:
If I go the WIN32 route, there's no recovery if it doesn't match... I think maybe keep as-is, just warn and not include the `realsense2-all` target?